pkgs/development/libraries/p4est-sc/default.nix

```diff
@@ -0,0 +1,71 @@
+{ lib, stdenv, fetchgit
+, which, gnum4, autoconf, automake, libtool, pkgconf
+, p4est-sc-debugEnable ? true, p4est-sc-mpiSupport ? true
+, mpi ? null, openmpi ? null, openssh ? null, zlib
+}:
+
+# we prefer MPICH over OpenMPI; call accordingly
+assert p4est-sc-mpiSupport -> mpi != null;
+assert p4est-sc-mpiSupport && mpi == openmpi -> openssh != null;
```

Comment:
mpi is an alias to openmpi.

Reply:
Thanks for your comments, they helped a lot. I've updated the PR accordingly.

> mpi is an alias to openmpi.

I saw that, however, I myself prefer using on overlay with mpi = mpich. I'd like p4est to work with both (or without mpi, obviously). With openmpi, running make check requires openssh. Conversely, with mpich, openssh is not required.

I'd like to keep the dependencies minimal in any case. Naively, I'd like to also not require packages in the argument list to default.nix in case mpi or openssh disappear somehow (unlikely, but I like to be prepared regardless). Can we keep this as is, or is there a stronger argument for removing the default null values in the argument list?

Reply:
> openssh is not required

it is not required when it is behind a lib.option(s). There is no need to allow it be overwritten to null. This is very discouraged for normal packages and only causes bugs. Please remove that.

Reply:
Cool, sounds good.